deepchecks/nlp/utils/text_data_plot.py

```diff
@@ -0,0 +1,290 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2023 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""A module containing utils for displaying information on TextData object."""
+from typing import List, Optional
+
+import numpy as np
+import pandas as pd
+import plotly.graph_objs as go
+from plotly.subplots import make_subplots
+
+from deepchecks.nlp.task_type import TaskType, TTextLabel
+from deepchecks.nlp.utils.text_properties import TEXT_PROPERTIES_DESCRIPTION
+from deepchecks.utils.dataframes import un_numpy
+from deepchecks.utils.distribution.plot import get_density
+from deepchecks.utils.plot import feature_distribution_colors
+from deepchecks.utils.strings import format_percent, get_docs_link
+
+__all__ = ['text_data_describe_plot']
+
+# The threshold to display the maximum number of labels on the label distribution pie chart and display
+# rest of the labels under "Others" category.
+MAX_NUM_LABEL_DISTRIBUTION_TO_SHOW = 6
+
+
+def _calculate_annoation_ratio(label, n_samples, is_mutli_label, task_type):
+
+    if label is None:
+        return format_percent(0)
+    if is_mutli_label or task_type == TaskType.TOKEN_CLASSIFICATION:
+        annotated_count = 0
+        for label_data in label:
+            annotated_count = annotated_count + 1 if len(label_data) > 0 and pd.isna(label_data).sum() == 0 \
+                              else annotated_count
+        return format_percent(annotated_count / n_samples)
+    else:
+        return format_percent(pd.notna(label).sum() / n_samples)
+
+
+def _generate_table_trace(n_samples, annotation_ratio, categorical_metadata, numerical_metadata,
+                          categorical_properties, numerical_properties):
+    data_cell = ['<b>Number of samples</b>', '<b>Annotation ratio</b>', '<b>Metadata categorical columns</b>',
+                 '<b>Metadata numerical columns</b>', '<b>Categorical properties</b>', '<b>Numerical properties</b>']
+    info_cell = [n_samples, annotation_ratio]
+
+    if categorical_metadata is None or len(categorical_metadata) == 0:
```

Comment:
All those if else are repetitive, would either put them in a for loop / util function

Reply:
Done and move to util function